[ENH] BEP001 - RepetitionTimeExcitation and RepetitionTimePreparation (#671)
bids-standard/bids-specification: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

| ContrastBolusIngredient | OPTIONAL              | [string][]                             | Active ingredient of agent. Values MUST be one of: IODINE, GADOLINIUM, CARBON DIOXIDE, BARIUM, XENON Corresponds to DICOM Tag 0018,1048.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |

> | RepetitionTimeExcitation | OPTIONAL             | [number][] | The time elapses (in seconds) between two successive excitations. The DICOM tag that best refers to this parameter is [(0018, 0080)](http://dicomlookup.com/lookup.asp?sw=Tnumber&q=(0018,0080)). This field may be used together with the `RepetitionTimePreparation` for certain use cases, such as [MP2RAGE](https://doi.org/10.1016/j.neuroimage.2009.10.002). Use `RepetitionTimeExcitation` (in combination with `RepetitionTimePreparation` if needed) for anatomy imaging data rather than `RepetitionTime` as it is already defined as the amount of time that it takes to acquire a single volume in the [task imaging data](#task-including-resting-state-imaging-data) section. |

Reviewer: I think we can use "interval" to mean a duration of time. "interval, in seconds," should be doubly clear in case somebody has another notion for interval.

```suggestion
| RepetitionTimeExcitation | OPTIONAL             | [number][] | The interval, in seconds, between two successive excitations. The DICOM tag that best refers to this parameter is [(0018, 0080)](http://dicomlookup.com/lookup.asp?sw=Tnumber&q=(0018,0080)). This field may be used together with the `RepetitionTimePreparation` for certain use cases, such as [MP2RAGE](https://doi.org/10.1016/j.neuroimage.2009.10.002). Use `RepetitionTimeExcitation` (in combination with `RepetitionTimePreparation` if needed) for anatomy imaging data rather than `RepetitionTime` as it is already defined as the amount of time that it takes to acquire a single volume in the [task imaging data](#task-including-resting-state-imaging-data) section. |
```

@patsycle @HenkMutsaerts Is this also supposed to be an array? Or does ASL only need RTP to be variable?

Reviewer 2: @effigies For ASL, we only need RTP to be an array. So RTE can stay a number. Thanks ;)

> | RepetitionTimePreparation | OPTIONAL            | [number][] or [array][] of [numbers][] | The period of time in seconds that it takes a preparation pulse block to re-appear at the beginning of the succeeding (essentially identical) pulse sequence block. The data type number may apply to files from any MRI modality concerned with a single value for this field. The data type array provides a value for each volume in a 4D dataset and should only be used when the volume timing is critical for interpretation of the data, such as in [ASL](#arterial-spin-labeling-perfusion-data).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |

Reviewer: ```suggestion
| RepetitionTimePreparation | OPTIONAL            | [number][] or [array][] of [numbers][] | The interval, in seconds, that it takes a preparation pulse block to re-appear at the beginning of the succeeding (essentially identical) pulse sequence block. The data type number may apply to files from any MRI modality concerned with a single value for this field. The data type array provides a value for each volume in a 4D dataset and should only be used when the volume timing is critical for interpretation of the data, such as in [ASL](#arterial-spin-labeling-perfusion-data).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```

| **Key name**            | **Requirement level** | **Data type** | **Description**                                                                                                                          |
| ----------------------- | --------------------- | ------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
| ContrastBolusIngredient | OPTIONAL              | [string][]    | Active ingredient of agent. Values MUST be one of: IODINE, GADOLINIUM, CARBON DIOXIDE, BARIUM, XENON Corresponds to DICOM Tag 0018,1048. |

> | RepetitionTimeExcitation | OPTIONAL              | [number][]    | The time in seconds between successive excitation pulses that excite a spin system. The DICOM tag that best refers to this parameter is [(0018, 0080)](http://dicomlookup.com/lookup.asp?sw=Tnumber&q=(0018,0080)). This field may be used together with the `RepetitionTimePreparation` for certain use cases, such as [MP2RAGE](https://infoscience.epfl.ch/record/172927/files/mp2rage.pdf). Use `RepetitionTimeExcitation` (in combination with `RepetitionTimePreparation` if needed) for anatomy imaging data rather than `RepetitionTime` as it is already defined as the amount of time that it takes to acquire a single volume in the [task imaging data](#task-including-resting-state-imaging-data) section.|

Reviewer 3: ```
[MP2RAGE](https://infoscience.epfl.ch/record/172927/files/mp2rage.pdf)
```

Detail I am not sure about because I don't think it is consistent through the specs, but isn't it better to have a DOI for references? 

I think we sort of have in mind to automate the "bibliography" part of the specs, no?

Author: Sure @Remi-Gau, I will replace them with DOI URLs.

Author: 1a9412c32bdb6aa14314b1e0df2cb63c70a81e59

> | RepetitionTimeExcitation | OPTIONAL. The time in seconds between successive excitation pulses that excite a spin system. The DICOM tag that best refers to this parameter is [(0018, 0080)](http://dicomlookup.com/lookup.asp?sw=Tnumber&q=(0018,0080)). This field may be used together with the `RepetitionTimePreparation` for certain use cases, such as [MP2RAGE](https://infoscience.epfl.ch/record/172927/files/mp2rage.pdf). Use `RepetitionTimeExcitation` (in combination with `RepetitionTimePreparation` if needed) for anatomy imaging data rather than `RepetitionTime` as it is already defined as the amount of time that it takes to acquire a single volume in the [task imaging data](#task-including-resting-state-imaging-data) section.|
> | RepetitionTimePreparation | OPTIONAL. The period of time in seconds that it takes a preparation pulse block to re-appear at the beginning of the succeeding (essentially identical) pulse sequence block. |

Reviewer: ```suggestion
| RepetitionTimeExcitation | OPTIONAL              | [number][]    | The time in seconds between successive excitation pulses that excite a spin system. The DICOM tag that best refers to this parameter is [(0018, 0080)](http://dicomlookup.com/lookup.asp?sw=Tnumber&q=(0018,0080)). This field may be used together with the `RepetitionTimePreparation` for certain use cases, such as [MP2RAGE](https://infoscience.epfl.ch/record/172927/files/mp2rage.pdf). Use `RepetitionTimeExcitation` (in combination with `RepetitionTimePreparation` if needed) for anatomy imaging data rather than `RepetitionTime` as it is already defined as the amount of time that it takes to acquire a single volume in the [task imaging data](#task-including-resting-state-imaging-data) section.|
| RepetitionTimePreparation | OPTIONAL              | [number][]    | The period of time in seconds that it takes a preparation pulse block to re-appear at the beginning of the succeeding (essentially identical) pulse sequence block. |
```
